apm/src/app/user/user.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { UserService } from './user.service';
+
+@Component({
+  templateUrl: './user.component.html',
+  styleUrls: ['./user.component.css']
+})
+export class UserComponent implements OnInit {
+
+  public editForm!: FormGroup;
+  private firstName!: FormControl;
+  private lastName!: FormControl;
+  private readonly homepageUrl = "/welcome";
+  constructor(private readonly userService: UserService, private readonly router: Router) { }
+  
+  public get FirstName() {
+    return this.firstName.value;
```

Comment:
Also you should always check for what you want to test against, like are you testing for a blank value or null or undefined. With the way it is we are still depending or thinking of implementation like it is dynamically typed